src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.State.cs

```diff
@@ -20,9 +19,11 @@ internal partial class AddConstructorParametersFromMembersCodeRefactoringProvide
         private class State
         {
             public ImmutableArray<ConstructorCandidate> ConstructorCandidates { get; private set; }
-            public INamedTypeSymbol ContainingType { get; private set; }
 
-            public static async Task<State> GenerateAsync(
+            [NotNull]
+            public INamedTypeSymbol? ContainingType { get; private set; }
```

Comment:
😕 Why?

Reply:
This type is factory constructed. If you get an instance of it, the factory will have checked that this was not null.

Reply:
Why not just `public INamedTypeSymbol ContainingType { get; }`?

Reply:
i got warnings then that the constructor doesn't set it.  Do we have a preferred idiom hre?  It's a transitional value.  but i would prefer it represent the value for the final consumer that passes this around and operates on it.